genomic_data_service/region_indexer_file_reader.py

```diff
@@ -10,13 +10,17 @@
 MAX_IN_MEMORY_FILE_SIZE = (25 * 1024 * 1024)
 
 
-class S3BedFileRemoteReader():
-    def __init__(self, file_properties, strand_col_values, snp_set=False):
-        self.file_properties = file_properties
+class FileReader():
+    def __init__(self, file_path, strand_col_values={}, file_size=0, source="s3"):
+        self.file_size = file_size
+        self.file_path = file_path
         self.temp_file = tempfile.NamedTemporaryFile()
-        self.snp_set = snp_set
         self.strand_values = strand_col_values
-        self.data = self.download_file_from_s3()
+        self.data = None
+        if source == "s3":
+            self.data = self.download_file_from_s3()
+        elif source == "local":
+            self.data = gzip.open(self.file_path, mode='rt')
```

Comment:
I think that any of the subclasses could either be on local disk or S3; "local" disk is just used for testing?
Making each subclass composed of a parse+downloader seems like a lot of effort to remove 1 if statement.

Reply:
I don't think the point of this effort is removing an if statement. The point is to work towards making the code readable and also making writing tests easier (without needing to hit external resources and/or mocking half of the AWS).

Reply:
Also, as we know the file formats do change and get added so while now keeping one if-statement might feel expedient will become a nightmare in the future if we just keep on hacking.

Reply:
I suppose instead of just complaining I can give an actionable suggestion. I think one quite low effort way of achieving the goals of being neat and pretty extendable would be to have the separate (sub)classes for reading from different sources, but not bothering making classes for reading different formats. Instead the parser function could be just passed in as a parameter to the class when an object is instantiated. The parsers would live in a separate module `parsers.py` where we can add more or modify without the reader classes ever needing to care. These parsers should be able to just take any iterable of tokens (lines) that it then can parse one by one. They could now be easily tested in separation without any mocks. @hitz Does this sound like a reasonable to you?

Reply:
@ottojolanki I am working on it.